orchagent/neighorch.cpp

```diff
@@ -865,3 +952,284 @@ bool NeighOrch::removeTunnelNextHop(const NextHopKey& nh)
+            }
+
+            if(!encap_index)
+            {
+                //Encap index is not available yet. Since this is remote neighbor, we need to wait till
+                //Encap index is made available either by dynamic syncing or by static config
+                it++;
+                continue;
+            }
+
+            if (m_syncdNeighbors.find(neighbor_entry) == m_syncdNeighbors.end() ||
+                    m_syncdNeighbors[neighbor_entry].mac != mac_address)
+            {
+                //Add neigh to SAI
+                if (addNeighbor(neighbor_entry, mac_address))
+                {
+                    //neigh successfully added to SAI. Set STATE DB to signal kernel programming by neighbor manager
+
+                    //If the inband interface type is not VLAN, same MAC can be used for the inband interface for
+                    //kernel programming.
+                    if(ibif.m_type != Port::VLAN)
+                    {
+                        mac_address = gMacAddress;
+                    }
+                    vector<FieldValueTuple> fvVector;
+                    FieldValueTuple mac("neigh", mac_address.to_string());
+                    fvVector.push_back(mac);
+                    m_stateSystemNeighTable->set(state_key, fvVector);
+
+                    it = consumer.m_toSync.erase(it);
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("Failed to add voq neighbor %s to SAI", kfvKey(t).c_str());
+                    it++;
+                }
+            }
+            else
+            {
+                /* Duplicate entry */
+                SWSS_LOG_INFO("System neighbor %s already exists", kfvKey(t).c_str());
+                it = consumer.m_toSync.erase(it);
+            }
+        }
+        else if (op == DEL_COMMAND)
+        {
+            if (m_syncdNeighbors.find(neighbor_entry) != m_syncdNeighbors.end())
+            {
+                //Remove neigh from SAI
+                if (removeNeighbor(neighbor_entry))
+                {
+                    //neigh successfully deleted from SAI. Set STATE DB to signal to remove entries from kernel
+                    m_stateSystemNeighTable->del(state_key);
+
+                    it = consumer.m_toSync.erase(it);
+                }
+                else
+                {
+                    SWSS_LOG_ERROR("Failed to remove voq neighbor %s from SAI", kfvKey(t).c_str());
+                    it++;
+                }
+            }
+            else
+                /* Cannot locate the neighbor */
+                it = consumer.m_toSync.erase(it);
+        }
+        else
+        {
+            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+            it = consumer.m_toSync.erase(it);
+        }
+    }
+}
+
+bool NeighOrch::addInbandNeighbor(string alias, IpAddress ip_address)
+{
+    //For "port" type inband, the inband reachability info syncing can be done through static
```

Comment:
Can we remove these functions if not used?

Reply:
This function and delInbandNeighbor() are place holders for forthcoming Inband PRs (port type by Nokia and vlan type by Arista). These dummy functions are there with some brief comments as guidance/remainder for filling these functions, when Inband interface functionality is implemented.